src/schema/fields/listPrice.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  GraphQLFieldConfig,
+  GraphQLObjectType,
+  GraphQLInt,
+  GraphQLUnionType,
+  GraphQLNonNull,
+} from "graphql"
+
+const PriceRange = new GraphQLObjectType({
+  name: "PriceRange",
+  fields: {
+    minPriceCents: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+    maxPriceCents: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+  },
+})
+
+const ExactPrice = new GraphQLObjectType({
+  name: "ExactPrice",
+  fields: {
+    priceCents: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+  },
+})
+
+export const listPrice: GraphQLFieldConfig<any, any> = {
+  type: new GraphQLUnionType({
+    name: "ListPrice",
+    types: [PriceRange, ExactPrice],
```

Comment:
do we need a `resolve_type` definition here? something like 
https://github.com/artsy/metaphysics/blob/7d38d29c8e245b0e78b2f5f16790ab3967981d5a/src/schema/ecommerce/types/order_party_union.ts#L8
i feel like with that you may not need to manually set the `__typename` in 
https://github.com/artsy/metaphysics/pull/1669/files#diff-c8c5574211ce46fb63e10901c34dfee3R43
and then each type could have a `resolve` getting `price_cents`. what do you think?

Reply:
`listPrice` still needs a resolver because it's deciding which of the union types to resolve for. So then we'd be moving the resolution logic to four different places (one for each field, listPrice, minPriceCents, maxPriceCents, and priceCents) instead of just one (listPrice), when the concrete types aren't being used independently. Is there any benefit in doing that?